### PR TITLE
resourcemanger: add cpu scheduler

### LINKED ISSUE #39657: add resource manage to schedule pool by static
## Enhancement

Create an resource manager to schedule globally pool. resource manager can tune the parallel degree of pool by adjusting worker number.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #39657

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: resourcemanager/scheduler/cpu_scheduler.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+func NewCPUScheduler() *CPUScheduler {
+	return &CPUScheduler{}
+}
+
+// Tune is to tune the goroutine pool
+func (*CPUScheduler) Tune(_ util.Component, pool util.GorotinuePool) Command {
+	if time.Since(pool.LastTunerTs()) < minCPUSchedulerInterval.Load() {
+		return Hold
+	}
+	if cpu.GetCPUUsage() < 0.5 {
```
Comment (Benjamin2037) [inline resourcemanager/scheduler/cpu_scheduler.go:37]: why get cpu usage < 0.5, downclock? I understand now there is more cpu resource, should overclock?

Comment (hawkingrei) [reply resourcemanager/scheduler/cpu_scheduler.go:37]: Yes, fixed.

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Comment by hawkingrei** [comment]
@zimulala @YangKeao PTAL

**Comment by hawkingrei** [comment]
@tangenta @tiancaiamao PTAL

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**